src/core/Ros.js

```diff
@@ -3,15 +3,17 @@
  * @author Brandon Alexander - baalexander@gmail.com
  */
 
-var WebSocket = require('ws');
-var WorkerSocket = require('../util/workerSocket');
-var socketAdapter = require('./SocketAdapter.js');
+import WebSocket from 'ws';
+// import {WorkerSocket} from '../util/workerSocket.js'; // TODO
```

Comment:
- [ ] TODO I disabled some shims to get it to compile for browsers first. These need to be re-shimmed using Rollup.

Also another question I have is, is WorkerSocket actually useful? Has there been performance gains from this? I'm skeptical because fetching from network is a non-blocking operation. Does the data handling (from the browser engine, before finally passing that data to JS) have a big cost that the worker is putting on a separate thread?

Reply:
I think (but I'm not sure) that the CPU heavy work is decompressing the messages. Suppose you are sending a pointcloud with png based message compression. This work will then happen in a worker.

Reply:
That makes sense, but unless I overlooked it, it seemed to me that only the WebSocket network request runs in the worker, and passes the data back to the main thread to process the PNG. I don't see that `decompressPng` file being imported in the worker, I just see it `postMessage`ing data.